### DISCUSSION TITLE
Some questions about grain_growth_3D.i

### DISCUSSION THREAD

**Discussion by PengWei97** [idaholab/moose#16473] [Q&A Modules: Phase field] [answered]
Dear MOOSE experts,
Recently, I am studying the grain growth model in the phase field. Based on the input file [grain_growth_3D.i](https://github.com/idaholab/moose/blob/next/modules/phase_field/examples/grain_growth/grain_growth_3D.i), I have some questions and hope to get your guidance and corrections.
The following is my understanding and confusion,

First of all, why is there no BCs block in grain_growth_3D.i?

Second, in AuxVariables and AuxKernels, apart from bnds,I do not understand what other auxiliary variables and kernels are used for? in other words:

- In AuxVariables, what are the meanings of unique_grains, var_indices, ghost_regions, and halos and what are their functions?
- In AuxKernels, what are the functions implemented by flood_counter = grain_tracker and flood_counter = voronoi? I did not find the relevant option in [FeatureFloodCountAux](https://mooseframework.inl.gov/source/postprocessors/FeatureFloodCount.html#grain_stitch_split).
- What does field_display = VARIABLE_COLORING mean in AuxKernels? Other options such as: What do VARIABLE_COLORING and GHOSTED_ENTITIES mean?
`[AuxKernels]
  [./var_indices]
    type = FeatureFloodCountAux
    variable = var_indices
    flood_counter = grain_tracker
    field_display = VARIABLE_COLORING
    execute_on = 'initial timestep_end'
  [../]
...`

Finally, the input file input is a *_out.e file, but the multiple input files output in grain_growth_2D_voronoi.i, what input code determines the extension of the .e result file input one or more?

Any suggestions or recommendations to fix these problems would be greatly appreciated.
Thank you
Wei Peng

**Comment by jiangwen84** [comment, marked as answer]
1. For FEM, if boundary condition is not specified, it implies a homogeneous Neumann boundary condition, i.e. flux = 0.

2. Your second question is related to grain tracker, and you can find more information on  https://mooseframework.inl.gov/source/postprocessors/GrainTracker.html#!

3. The input file should always be .i file, and *_out.e are the output file that can be visualized in Paraview or Peacock.

**Reply by PengWei97** [reply]
Thank you for your answers. For the third question, I may not have made it clear. After the grain_growth_3D.i simulation is completed, only one out.e file is generated. However, grain_growth_2D_voronoi.i simulation generated hundreds of out.e files. So my question is what code in the input file affects the generation of one or more out.e files?

**Reply by SudiptaBiswas** [reply]
Multiple output files are generated when the adaptive mesh is used. With adaptivity, every time the mesh is changed it creates a new output file.

**Reply by PengWei97** [reply]
Ok,thanks for your answers.

**Reply by permcody** [reply]
Stacking lots of questions into a single post may mean you don't get satisfactory answers to every question. Let us know if you have others that weren't answered here.
